src/clojure/nrepl/util/completion.clj

```diff
@@ -168,16 +168,18 @@
   (.. file (replace ".class" "") (replace File/separator ".")))
 
 (def top-level-classes
-  (misc/noisy-future
-   (doall
-    (for [file classfiles :when (re-find #"^[^\$]+\.class" file)]
-      (classname file)))))
+  (delay
```

Comment:
Just add some note about the `delay`s, otherwise someone might optimize them away. :-)

Reply:
Done @bbatsov